### PR TITLE
[ci] set `--tag` on release

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- We currently don't have a process for backporting fixes to previous release branches
- We _do_ currently release from the `main`, `next`, and `1-legacy` / `2-legacy` / `3-legacy` branches
- When we recently cut a release from `3-legacy`, it was published to `latest` on NPM! Thankfully we released from `main` shortly thereafter, but it could be annoying to debug if we forget to do that in the future.
- This PR automates the process so we don't have to think about it again. The code comments hopefully provide enough context.
- Closes PLT-1750

## Testing

Contrary to my typical YOLO approach, I actually minimized this workflow and tested it in a private repo. Everything works as expected!

## Docs

Currently, our actions are super undocumented. Currently attempting to kick off an effort to get more of these decisions documented, but until then, code comments should suffice.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -21,7 +21,6 @@ env:
 jobs:
   changelog:
     name: Changelog PR or Release
-    if: ${{ github.repository_owner == 'withastro' }}
     runs-on: ubuntu-latest
```
Comment (natemoo-re) [inline .github/workflows/release.yml:24]:
```suggestion
		if: ${{ github.repository_owner == 'withastro' }}
    runs-on: ubuntu-latest
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -21,7 +21,6 @@ env:
 jobs:
   changelog:
     name: Changelog PR or Release
-    if: ${{ github.repository_owner == 'withastro' }}
```
Comment (bluwy) [inline .github/workflows/release.yml:24]: Is there a reason this is moved into a step below? Shouldn't we skip this workflow entirely on forks?

**Review by ematipico — APPROVED** [review]
Looks good to me

File: .github/workflows/release.yml (hunk 2 above)
Comment (natemoo-re) [reply .github/workflows/release.yml:24]: Didn't mean to change this, but yes, can totally skip the whole workflow.

**Pull request merged**